client/allocrunner/taskrunner/task_runner_getters.go

```diff
@@ -72,7 +72,7 @@ func (tr *TaskRunner) setVaultToken(token string) {
 	// Update the task's environment
 	taskNamespace := tr.task.Vault.Namespace
 
-	ns := tr.clientConfig.VaultConfig.Namespace
+	ns := tr.clientConfig.VaultConfigs[tr.task.GetVaultClusterName()].Namespace
```

Comment:
```suggestion
	ns := tr.clientConfig.GetVaultConfigs(tr.logger)[tr.task.GetVaultClusterName()].Namespace
```

I guess that, at this point, it's safe to assume the client config exists?

Reply:
Yeah, `clientConfig` has to be set to create the taskrunner. And the `VaultConfig` we get should always be non-nil because the fingerprint from that config is what tells the server it can place this alloc on this node.

Reply:
Oh no! But if the user _removes_ a Vault config after the allocation has been placed and then restarts the client, the client would potentially crash during restore instead of just killing the non-working allocation. Shoot, I'm going to have to rework this approach a good bit.

Reply:
I'm going to pull that part of the work into https://github.com/hashicorp/nomad/issues/19000 just to keep this PR a manageable size for you to review @lgfa29

Reply:
Ah good catch!